build_python.sh: Allow to install multiple python packages with --extra_packages

enkiusz opened the pull request:
Before:

```
$ scripts/build_python.sh --extra_packages 'requests structlog requests_toolbelt' -i venv
[...]
Done. Made 10372 targets from 433 files in 189ms
ninja: Entering directory `./out/python_lib'
ninja: no work to do.
Creating a clear VirtualEnv in 'venv' ...
Looking in links: /tmp/tmpcos5g_up
Requirement already satisfied: pip in ./venv/lib/python3.12/site-packages (24.0)
ERROR: Invalid requirement: 'requests structlog requests_toolbelt'
$ 
```

After:

```
$ scripts/build_python.sh --extra_packages 'requests structlog requests_toolbelt' -i venv
[...]
[849/849] ACTION //src/controller/python:chip-repl(//build/toolchain/linux:linux_x64_gcc)
Creating a clear VirtualEnv in 'venv' ...
Looking in links: /tmp/tmpctksndlt
Requirement already satisfied: pip in ./venv/lib/python3.12/site-packages (24.0)
Processing ./out/python_lib/controller/python/chip_clusters-0.0-py3-none-any.whl
Processing ./out/python_lib/controller/python/chip_core-0.0-cp37-abi3-linux_x86_64.whl
Processing ./out/python_lib/controller/python/chip_repl-0.0-py3-none-any.whl
Processing ./out/python_lib/obj/src/python_testing/matter_testing_infrastructure/chip-testing._build_wheel/chip_testing-1.4.1-py3-none-any.whl
Processing ./out/python_lib/obj/scripts/matter_yamltests_distribution._build_wheel/matter_yamltests-0.0.1+20250326133422-py3-none-any.whl
Collecting requests
  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting structlog
  Using cached structlog-25.2.0-py3-none-any.whl.metadata (8.6 kB)
Collecting requests_toolbelt
  Using cached requests_toolbelt-1.0.0-py2.py3-none-any.whl.metadata (14 kB)
$ 
```

#### Testing

Tested by running the above command on master and confirming that the venv builds with the requested packages and works (`chip` can be imported).

enkiusz committed:
build_python.sh: Allow to install multiple python packages with --extra_packages

Signed-off-by: Maciej Grela <m.grela@samsung.com>

andy31415 commented on scripts/build_python.sh:
```diff
@@ -190,7 +190,7 @@ if [ "$install_pytest_requirements" = "yes" ]; then
 
 if [ -n "$extra_packages" ]; then
-    WHEEL+=("$extra_packages")
+    WHEEL+=($extra_packages)
```
looks good, however I believe restyler shellharden fights you (and probably what got this to be wrong in the first place...).

Ideally we would find some other way that works, but if that fails we can mark this in ignore at https://github.com/project-chip/connectedhomeip/blob/master/.restyled.yaml#L41

Please try to see if you find a method that both works and makes shellharden happy ... and if that fails lets add the exception.

enkiusz replied:
Yes I have seen that, I will look into this tomorrow.

enkiusz replied:
Changed the script so that extra_packages can be specified multiple times and uses an array. I was tempted to rename the option to --extra_package in order to confer the change in meaning but that could break for ppl already using it.

enkiusz committed:
Merge branch 'project-chip:master' into python-build-allow-multiple-extra-pkgs

enkiusz committed:
build_packages.sh: Improve the way multiple extra PyPI packages can be defined

Currently restyle's shellharden does not allow us to use shell word splitting to append
multiple package names to WHEEL. In order to fix this and keep restyled happy the
--extra_packages can now be specified multiple times collecting the extra packages into
an array. A shorthand was also added to make this less verbose.

Signed-off-by: Maciej Grela <m.grela@samsung.com>
